File: 1erdepartamental/python/14.recursividad/main.py
"""
Programa: Recursividad
Demuestra algoritmos recursivos clásicos
"""
#Clase Principa
from core.algoritmos import AlgoritmosRecursivos

def main ():
    algo = AlgoritmosRecursivos()
    print(f"Factorial de 5: {algo.factorial(5)}")
    print(f"Fibonacci de 7: {algo.fibonacci(7)}")
    print(f"Suma hasta 10: {algo.suma_hasta(10)}")
    print(f"2^8: {algo.potencia(2, 8)}")
    
if __name__ == "__main__":
   main()

File: 1erdepartamental/c++/14.recursividad/main.cpp
#include <iostream>
#include "core/AlgoritmosRecursivos.h"

int main() {
    AlgoritmosRecursivos algo;
    
    std::cout << "Factorial de 5: " << algo.factorial(5) << std::endl;
    std::cout << "Fibonacci de 7: " << algo.fibonacci(7) << std::endl;
    std::cout << "Suma hasta 10: " << algo.sumaHasta(10) << std::endl;
    std::cout << "2^8: " << algo.potenciar(2, 8) << std::endl;
    
    return 0;
}
